src/Processors/Formats/Impl/ParquetBlockInputFormat.cpp

```diff
@@ -418,6 +418,21 @@ void ParquetBlockInputFormat::initializeIfNeeded()
     int num_row_groups = metadata->num_row_groups();
     row_group_batches.reserve(num_row_groups);
 
+    auto adative_chunk_size = [&](int row_group_idx) -> size_t
+    {
+        size_t total_size = 0;
+        auto row_group_meta = metadata->RowGroup(row_group_idx);
+        for (int column_index : column_indices)
+        {
+            total_size += row_group_meta->ColumnChunk(column_index)->total_uncompressed_size();
+        }
+        if (!total_size || !format_settings.parquet.prefer_block_bytes) return 0;
+        auto average_row_bytes = total_size / row_group_meta->num_rows();
+        /// max_block_size >= num_rows >= 128
+        auto num_rows = std::min(format_settings.parquet.prefer_block_bytes/average_row_bytes, format_settings.parquet.max_block_size);
+        return std::max(num_rows, 128UL);
```

Comment:
One of my ideas here is to prevent users from setting too low block rows, which will seriously affect performance and have no practical effect.

Reply:
> which will seriously affect performance and have no practical effect

I agree on the performance, however this would be something new that was not present before. Somebody, somewhere might depend on setting max rows to less than 128.

Furthermore if you would like to do something like this, it is better to do it for [multiple settings](https://github.com/ClickHouse/ClickHouse/blob/138f5813bbe69b8af46aca60b770204d33b99080/src/Core/SettingsQuirks.cpp#L112-L130).